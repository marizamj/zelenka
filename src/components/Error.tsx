import * as React from "react";
import "../css/Error.css";

const Error = () => (
  <div className="error" data-testid="error">
    Что-то пошло не так. Пожалуйста, напишите нам на почту{" "}
    <a href="mailto:zelenkanasha@gmail.com">zelenkanasha@gmail.com</a> или в{" "}
    <a
      href="https://www.facebook.com/3elenka.online/"
      target="_blank"
      rel="noopener noreferrer"
    >
      фейсбук
    </a>
    .
  </div>
);

export default Error;
