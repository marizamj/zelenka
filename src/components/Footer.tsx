/** @jsx jsx */
import { jsx, css } from "@emotion/core";

const Footer = ({ currentYear }: { currentYear?: number }) => (
  <footer css={footerStyle} data-testid="footer">
    &#169; 2018-{currentYear || new Date().getFullYear()} zelenka.online{" "}
    <a href="mailto:zelenkanasha@gmail.com">zelenkanasha@gmail.com</a>
  </footer>
);

const footerStyle = css`
  border-top: 1px solid #398a85;
  padding: 1em;
  margin-top: 2em;
`;

export default Footer;
