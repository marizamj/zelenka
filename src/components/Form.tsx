/** @jsx jsx */
import { jsx, css } from "@emotion/core";
import { useState, FormEvent, useRef } from "react";
import formFields from "../lib/formFields";
import { breakpoints } from "../lib/misc";
import { FormType, FormData, FormFull } from "../types";

interface Props {
  type: FormType;
  handleSubmit: (formState: FormData) => void;
}

const Form = ({ type, handleSubmit }: Props) => {
  const [formState, setFormState] = useState<FormFull>({} as FormFull);
  const [submitAttempted, setSubmitAttempted] = useState(false);
  const errorMessage = useRef<HTMLParagraphElement>(null);

  const preventDefaultInvalidField = (
    e: FormEvent<HTMLInputElement | HTMLTextAreaElement>
  ) => e.preventDefault();

  const findFirstInvalidField = () =>
    Array.from<HTMLInputElement | HTMLTextAreaElement>(
      document.querySelectorAll("[id^=field][required]")
    ).find(el => !el.value);

  return (
    <form
      css={formStyle}
      onInvalid={e => {
        setSubmitAttempted(true);
        window.scrollTo({ top: errorMessage.current?.offsetTop });
        findFirstInvalidField()?.focus();
      }}
      onSubmit={e => {
        e.preventDefault();
        handleSubmit(formState);
      }}
      data-testid={`form-${type}`}
    >
      <div css={formDescriptionStyle}>{formFields[type].description}</div>
      <p css={requiredStyle} ref={errorMessage}>
        * - обязательные поля
      </p>
      {submitAttempted && (
        <p css={fillRequiredFieldsStyle}>
          Пожалуйста, заполните все обязательные поля.
        </p>
      )}
      {formFields[type].fields.map(({ label, required, name, type }) => (
        <label
          css={[
            formLabelStyle,
            submitAttempted &&
              required &&
              !formState[name] &&
              formLabelErrorStyle
          ]}
          key={label}
          htmlFor={`field-${name}`}
        >
          <p>
            {label}
            {required && <span css={requiredStyle}>*</span>}
          </p>
          {type === "textarea" ? (
            <textarea
              onChange={({ target: { value, name } }) => {
                setFormState({ ...formState, [name]: value });
              }}
              onInvalid={preventDefaultInvalidField}
              required={required}
              name={name}
              id={`field-${name}`}
              data-testid={`field-${name}`}
              value={formState[name] || ""}
            />
          ) : (
            <input
              onChange={({ target: { value, name } }) => {
                setFormState({ ...formState, [name]: value });
              }}
              onInvalid={preventDefaultInvalidField}
              required={required}
              type={type || "text"}
              name={name}
              id={`field-${name}`}
              data-testid={`field-${name}`}
              value={formState[name] || ""}
            />
          )}
        </label>
      ))}
      <input type="submit" value="Отправить" data-testid="submit" />
    </form>
  );
};

const formStyle = css`
  font-size: 1.5em;
  width: 50%;
  margin-left: 25%;
  text-align: center;
  input,
  textarea {
    width: calc(100% - 20px);
    padding: 10px;
    outline: none;
    border: 1px solid lightgrey;
    border-radius: 3px;
    font-size: 0.8em;
  }
  textarea {
    height: 4em;
    resize: none;
  }
  input:focus,
  textarea:focus {
    border: 1px solid grey;
  }
  input[type="submit"] {
    cursor: pointer;
    color: dimgrey;
    font-size: 1em;
    width: auto;
    border: 1px solid lightgrey;
    border-radius: 3px;
    text-align: center;
    margin-top: 1em;
  }
  input[type="submit"]:hover {
    background: #f5f5f5;
  }
  input:disabled {
    background: lightgrey;
    cursor: default;
  }
  ${breakpoints.maxWidth1023} {
    width: 80%;
    margin: 0;
    margin-left: 10%;
    font-size: 1.3em;
  }
`;

const requiredStyle = css`
  font-size: 0.8em;
  color: #c92121;
  text-align: end;
  margin-left: 0.1em;
`;

const fillRequiredFieldsStyle = css`
  text-align: start;
  font-size: 0.8em;
  color: #c92121;
`;

const formDescriptionStyle = css`
  text-align: center;
`;

const formLabelStyle = css`
  p {
    color: dimgrey;
    margin-top: 1.5em;
    margin-bottom: 0.5em;
    font-size: 0.8em;
    text-align: start;
  }
`;

const formLabelErrorStyle = css`
  p {
    color: #c92121;
  }
  *:focus {
    outline: none;
    box-shadow: 0 0 0 1px white, 0 0 0 4px #c92121;
    border-radius: 2px;
    transition: box-shadow 0.2s ease;
  }
`;

export default Form;
