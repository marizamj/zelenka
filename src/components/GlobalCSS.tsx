/** @jsx jsx */
import { jsx, css, Global } from "@emotion/core";

const GlobalCSS = () => <Global styles={globalStyle} />;

const globalStyle = css`
  body {
    margin: 0;
    padding: 0;
    font-family: "PT Sans", Arial, sans-serif;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    min-width: 320px;
  }

  code {
    font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New",
      monospace;
  }

  h1,
  h2 {
    color: #c92121;
    margin: 100px 0px 50px 0px;
  }

  *:focus {
    outline: none;
    box-shadow: 0 0 0 1px white, 0 0 0 4px #398a85;
    border-radius: 2px;
    transition: box-shadow 0.2s ease;
  }
`;

export default GlobalCSS;
