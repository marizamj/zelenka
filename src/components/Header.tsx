import * as React from "react";
import "../css/Header.css";
import { Link } from "react-router-dom";

import logo from "../images/logo.png";

const Header = () => (
  <header className="header">
    <div className="title-img" />
    <div className="logo-container">
      <Link to="/">
        <img src={logo} className="logo" alt="ЗЕЛЕНКА" />
      </Link>
      <p className="logo-text">
        Показываем хирургию и медицину так, чтобы не тошнило
      </p>
    </div>
  </header>
);

export default Header;
