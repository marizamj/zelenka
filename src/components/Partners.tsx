/** @jsx jsx */
import { jsx, css } from "@emotion/core";
import partners from "../lib/partners";
import { reactKey, breakpoints } from "../lib/misc";

const Partners = () => (
  <div css={partnersStyle}>
    <h2>Уже работают с нами</h2>

    {partners.map(({ id, title, url, img }) => (
      <div
        css={partnerContainerStyle}
        key={reactKey({ id, title, url, img })}
        role="link"
        onClick={() => window.open(url, "_blank")}
        tabIndex={0}
      >
        <img src={img} alt={title} css={partnerLogoStyle} />
        <span css={partnerNameStyle}>{title}</span>
      </div>
    ))}
  </div>
);

const partnersStyle = css`
  text-align: center;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-evenly;
  h2 {
    flex: 1 0 100%;
    margin-bottom: 1em;
  }
  ${breakpoints.maxWidth1023} {
    h2 {
      margin-bottom: 2em;
    }
  }
  ${breakpoints.maxWidth700} {
    display: initial;
  }
`;

const partnerContainerStyle = css`
  position: relative;
  padding: 2em 2em 4em 2em;
  margin-bottom: 2em;
  cursor: pointer;

  &:hover {
    span {
      text-decoration: underline;
    }
  }
`;

const partnerNameStyle = css`
  color: inherit;
  position: absolute;
  bottom: 0;
  left: 0;
  width: 100%;
  font-size: 1em;
`;

const partnerLogoStyle = css`
  cursor: pointer;
  height: 130px;
`;

export default Partners;
