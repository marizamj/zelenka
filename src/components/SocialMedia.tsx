/** @jsx jsx */
import { jsx, css } from "@emotion/core";
import smBackImg from "../images/sm-back.png";
import facebook from "../images/facebook.png";
// import instagram from "../images/instagram.png";
import youtube from "../images/youtube.png";
import { breakpoints } from "../lib/misc";

const SocialMedia = () => (
  <div css={smStyle}>
    <a
      href="https://www.youtube.com/channel/UCIxnXifkJRTPOIe4AQM_fBQ"
      target="_blank"
      rel="noopener noreferrer"
    >
      <img
        src={youtube}
        alt="YouTube"
        css={smBtnStyle}
        style={{ marginRight: "5.5em" }}
      />
    </a>
    {/* <a
          href="https://www.youtube.com/channel/UCIxnXifkJRTPOIe4AQM_fBQ"
          target="_blank"
          rel="noopener noreferrer"
        >
          <img
            src={instagram}
            alt="Instagram"
            css={smBtnStyle}
            style={{ marginRight: "12vw" }}
          />
        </a> */}
    <a
      href="https://www.facebook.com/3elenka.online/"
      target="_blank"
      rel="noopener noreferrer"
    >
      <img src={facebook} alt="Facebook" css={smBtnStyle} />
    </a>
  </div>
);

const smStyle = css`
  font-size: 2em;
  position: relative;
  background: url(${smBackImg});
  background-repeat: no-repeat;
  background-size: contain;
  background-position: right;
  height: 250px;
  ${breakpoints.maxWidth1023} {
    font-size: 1.5em;
    height: 190px;
    margin-top: -20px;
    margin-left: 0;
  }
`;

const smBtnStyle = css`
  height: 2em;
  position: absolute;
  top: 130px;
  right: 0;
  margin-right: 3em;
  cursor: pointer;
  &:hover {
    opacity: 0.8;
  }
  ${breakpoints.maxWidth1023} {
    top: 100px;
  }
`;

export default SocialMedia;
