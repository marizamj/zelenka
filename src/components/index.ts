export { default as Error } from "./Error";
export { default as Form } from "./Form";
export { default as Header } from "./Header";
export { default as Partners } from "./Partners";
export { default as SocialMedia } from "./SocialMedia";
export { default as ThankYou } from "./ThankYou";
export { default as Video } from "./Video";
