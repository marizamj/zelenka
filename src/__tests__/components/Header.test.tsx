import React from "react";
import { render } from "@testing-library/react";

import { Header } from "../../components";

describe("Header", () => {
  test("Matches snapshot", () => {
    const { container } = render(<Header />);

    expect(container).toMatchSnapshot();
  });
});
