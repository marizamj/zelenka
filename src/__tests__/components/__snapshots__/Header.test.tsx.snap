// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Matches snapshot 1`] = `
<div>
  <header
    class="header"
  >
    <div
      class="title-img"
    />
    <div
      class="logo-container"
    >
      <img
        alt="ЗЕЛЕНКА"
        class="logo"
        src="logo.png"
      />
      <p
        class="logo-text"
      >
        Показываем хирургию и медицину так, чтобы не тошнило
      </p>
    </div>
  </header>
</div>
`;
